lib/CachingOperation.ts

```diff
@@ -10,4 +12,27 @@ export class CachingOperation<LoadedValue> extends AbstractFlatOperation<LoadedV
       })
     }
   }
+  public setAsync(key: string, newValueFn: () => Promise<LoadedValue>): Promise<LoadedValue> {
+    const existingLoad = this.runningSetAsyncLoads.get(key)
+    if (existingLoad) {
+      return existingLoad
+    }
+
+    const resolverAndSetPromise = new Promise<LoadedValue>((resolve, reject) => {
+      const cacheValueResolver = newValueFn()
+      cacheValueResolver
+        .then((resolvedValue) => {
+          this.runningSetAsyncLoads.delete(key)
+          this.set(key, resolvedValue).finally(() => {
+            resolve(resolvedValue)
+          })
+        })
+        .catch((err) => {
+          this.runningSetAsyncLoads.delete(key)
+          reject(err)
```

Comment:
you can simply throw inside the promise chain instead of explicit rejection, when not using a manually created promise